tests/cpp-runtime/opencl/clml_memory_planner.cc

```diff
@@ -0,0 +1,481 @@
+    clReleaseMemObject(it->first);
+  }
+  cws->ddr_global_pool.clear();
+}
+
+void PlanMemory(tvm::runtime::contrib::CachedLayer& layer, int total_nodes,
+                std::map<int, uint32_t>& tensor_sizes,
+                std::map<int, std::vector<int>>& input_tensors) {
+  tvm::runtime::contrib::CLMLWorkspace* cws = tvm::runtime::contrib::CLMLWorkspace::Global();
+
+  for (int nid = 0; nid < total_nodes; ++nid) {
+    uint32_t size = tensor_sizes[nid];
+    size_t offset = -1;
+    if (cws->is_on_chip_memory) {
+      LOG(WARNING) << "Requesting On-chip:" << nid;
+      offset = RequestOnChipMemory(&layer, size);
+    }
+    if (-1 != offset) {
+      LOG(WARNING) << "On Chip not found:" << nid;
+      layer.on_chip_pool_alloc_info.insert({offset, nid});
+      layer.on_chip_alloc_plan.insert({nid, std::make_pair(size, offset)});
+    } else {
+      LOG(WARNING) << "Requesting DDR memory:" << nid;
+      layer.on_chip_reject.insert({nid, size});
+      // DDR Allocation
+      auto ddr_mem = RequestDDRMemory(&layer, size);
+      layer.ddr_alloc_plan.insert({nid, ddr_mem});
+    }
+
+    // Now free up the input tensors on-chip memory for reuse.
+    for (auto& input_node : input_tensors[nid]) {
+      FreeMemory(&layer, input_node);
+    }
+  }
+
+#if 0
+  // Stats dump
+  size_t in_chip_total_alloc = 0;
+  size_t total_reject = 0;
+  for (auto it = layer.on_chip_alloc_plan.begin(); it != layer.on_chip_alloc_plan.end(); it++) {
+    LOG(WARNING) << " On-chip Alloc:" << it->first << " Size:" << it->second.first
+              << " Offset:" << it->second.second;
+    in_chip_total_alloc += it->second.first;
+  }
+
+  for (auto it = layer.on_chip_reject.begin(); it != layer.on_chip_reject.end(); it++) {
+    LOG(WARNING) << "Reject:" << it->first << " Size:" << it->second;
+    total_reject += it->second;
+  }
+  LOG(WARNING) << "Total On-chip Alloc:" << in_chip_total_alloc + total_reject
+            << " On-Chip:" << in_chip_total_alloc << " Reject:" << total_reject;
+
+  for (auto it = cws->ddr_global_pool.begin(); it != cws->ddr_global_pool.end(); it++) {
+    LOG(WARNING) << "DDR Global pool - size:" << it->second.first << " Ref:" << it->second.second;
+  }
+  for (auto it = layer.ddr_storage_ref_map.begin();
+       it != layer.ddr_storage_ref_map.end(); it++) {
+    LOG(WARNING) << "DDR Local pool - size:" << it->second.first << " Ref cnt:" << it->second.second;
+  }
+#endif
+}
+
+void CompareOnChipPlan(tvm::runtime::contrib::CachedLayer& layer,
+                       const std::vector<int>& on_chip_plan) {
+  for (auto& nid : on_chip_plan) {
+    EXPECT_EQ(layer.on_chip_alloc_plan.find(nid) == layer.on_chip_alloc_plan.end(), false);
+  }
+}
+
+void CompareDDRPlan(tvm::runtime::contrib::CachedLayer& layer, const std::vector<int>& ddr_plan) {
+  for (auto& nid : ddr_plan) {
+    EXPECT_EQ(layer.ddr_alloc_plan.find(nid) == layer.ddr_alloc_plan.end(), false);
+  }
+}
+
+TEST(CLMLMemoryPlanner, sequential_all_on_chip) {
```

Comment:
You can use `TEST_F` instead of `TEST`. In this case, you can move all common parts to the test class. As an example, you can take a look at: https://github.com/apache/tvm/blob/main/tests/cpp-runtime/opencl/opencl_compile_to_bin.cc